fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingMultiTargetByRecords.java

```diff
@@ -102,20 +110,45 @@ CompletableFuture<Void> buildIndexInternalAsync() {
                                 SubspaceProvider subspaceProvider = common.getRecordStoreBuilder().getSubspaceProvider();
                                 // validation checks, if any, will be performed here
                                 return subspaceProvider.getSubspaceAsync(context)
-                                        .thenCompose(subspace -> buildMultiTargetIndex(subspaceProvider, subspace, null, null));
+                                        .thenCompose(subspace -> buildMultiTargetIndex(subspaceProvider, subspace));
                             })
                 ), common.indexLogMessageKeyValues("IndexingMultiTargetByRecords::buildIndexInternalAsync"));
     }
 
     @Nonnull
-    private CompletableFuture<Void> buildMultiTargetIndex(@Nonnull SubspaceProvider subspaceProvider, @Nonnull Subspace subspace, @Nullable byte[] start, @Nullable byte[] end) {
+    private CompletableFuture<Void> buildMultiTargetIndex(@Nonnull SubspaceProvider subspaceProvider, @Nonnull Subspace subspace) {
+        final TupleRange tupleRange = common.computeRecordsRange();
+        final byte[] rangeStart;
+        final byte[] rangeEnd;
+        if (tupleRange == null) {
+            rangeStart = rangeEnd = null;
+        } else {
+            final Range range = tupleRange.toRange();
+            rangeStart = range.begin;
+            rangeEnd = range.end;
+        }
+
+        final CompletableFuture<FDBRecordStore> maybePresetRangeFuture =
+                rangeStart == null ?
+                CompletableFuture.completedFuture(null) :
+                buildCommitRetryAsync((store, recordsScanned) -> {
+                    // Here: only records inside the defined records-range are relevant to the index. Hence, the completing range
+                    // can be preemptively marked as indexed.
+                    final RangeSet rangeSet = new RangeSet(store.indexRangeSubspace(common.getIndex()));
```

Comment:
Is it correct that only `common.getIndex()`'s index range subspace gets updated? That doesn't seem right for non-idempotent indexes, but maybe those are excluded from multi-target index builds?

Reply:
That's a bug. Will fix.

Reply:
✅